src/ucp/core/ucp_worker.c

```diff
@@ -980,18 +1013,24 @@ static ucs_status_t ucp_worker_add_resource_ifaces(ucp_worker_h worker)
 
     if (!ctx_tl_bitmap) {
         /* Context bitmap is not set, need to select the best tl resources */
-        tl_bitmap = 0;
-        status    = ucp_worker_select_best_ifaces(worker, &tl_bitmap);
+        tl_bitmap          = 0;
+        scalable_tl_bitmap = 0;
+        status             = ucp_worker_select_best_ifaces(worker, &tl_bitmap,
+                                                           &scalable_tl_bitmap);
         if (status != UCS_OK) {
             return status;
         }
         ucs_assert(tl_bitmap);
 
         /* Cache tl_bitmap on the context, so the next workers would not need
          * to select best ifaces. */
-        context->tl_bitmap = tl_bitmap;
+        context->tl_bitmap          = tl_bitmap;
         ucs_debug("Selected tl bitmap: 0x%lx (%d tls)",
                   tl_bitmap, ucs_popcount(tl_bitmap));
+
+        context->scalable_tl_bitmap = scalable_tl_bitmap;
```

Comment:
ned to fill `scalable_tl_bitmap` for non-unified mode as well. With non-unified mode `context->tl_bitmap` is initialized with all available transports and looks like `selected_tl_bitmap` just copies it.
`ucp_worker_select_best_ifaces` is supposed to be used for unified mode only.

Reply:
done

Reply:
how did you fix it? Looks like the scalability bitmap is the same as regular one in non-unified mode

Reply:
yes, it is just set to `tl_bitmap` rvalue for non-unified mode
do you think that we could remove not scalable enough from the bitmap in non-unified mode?

Reply:
Unified mode is disabled by default, so we have to build scalable bitmap for non-unified mode as well (maybe using some common function). bitmap approach sounds better, because you check each transport for scalability just once (and save it in the bitmap)